src/Functional/functions.php

```diff
@@ -800,3 +803,37 @@ function sequence($monads)
 {
     return traverse(identity, toTraversable($monads));
 }
+
+/**
+ * filterM :: Monad m => (a -> m Bool) -> [a] -> m [a]
+ *
+ * @param callable $f (a -> m Bool)
+ * @param array|Traversable $collection [a]
+ * @return Monad m [a]
+ */
+function filterM(callable $f, $collection)
+{
+    /** @var Monad $monad */
+    $monad = $f(head($collection));
+
+    $_filterM = function($collection) use($monad, $f, &$_filterM){
+        if(count($collection) == 0) {
+            return $monad->of([]);
```

Comment:
Since `of` is a static method, better way is to call it like so `$monad::of`
In some php version this is recored as an warning.